src/content/guides/packages.md

```diff
@@ -178,6 +178,78 @@ due to conflicting version constraints in the pubspec.
 To identify out-of-date packages that require editing the pubspec,
 use [`dart pub outdated`][outdated].
 
+## Getting dependencies in production - `pub get --enforce-lockfile`
+
+In some cases `dart pub get` will not give you the exact versions locked in
+`pubspec.lock`. For example:
+
+* If new dependencies are added to `pubspec.yaml` since `pubspec.lock` was created.
+* If the locked version no longer exists in the package repository.
+* If you upgraded the Dart SDK, and some packages no longer are compatible with
+  the sdk.
+  
+In all these cases `dart pub get` will
+
+* Unlock as needed.
+* Notify about any changes in the output.
+
+For example after adding `retry: ^3.0.0` to your dependencies:
+```console
+$ dart pub get
+Resolving dependencies... (1.0s)
+Downloading packages... 
++ retry 3.1.2
+```
+
+Also if the hash of a published version does not match the hash in
+`pubspec.lock`, `pub` will warn you, but fix `pubspec.lock` to reflect the
+published version:
```

Comment:
How about we cross-link to the glossary entry about pub content hashes?

```suggestion
Also, if the [content hash](/tools/pub/glossary#content-hashes) of a 
published package version differs from the hash in the `pubspec.lock` file,
`pub` will warn you and update the lockfile to reflect the published version:
```

Reply:
Yeah!